platform/ios/src/MGLMapView.mm

```diff
@@ -3970,6 +3970,11 @@ - (void)setSelectedAnnotations:(NS_ARRAY_OF(id <MGLAnnotation>) *)selectedAnnota
 
 - (void)selectAnnotation:(id <MGLAnnotation>)annotation animated:(BOOL)animated
 {
+    [self selectAnnotation:annotation forGestureRegognizer:nil animated:animated];
+}
+
+- (void)selectAnnotation:(id <MGLAnnotation>)annotation forGestureRegognizer:(UITapGestureRecognizer *)singleTap animated:(BOOL)animated
```

Comment:
`singleTap` is being used only to determine where the callout should be anchored. This method should take a `positioningRect` explicitly: `-selectAnnotation:animated:calloutPositioningRect:`. The existing code that determines `positioningRect` can live in `-selectAnnotation:animated:`.